packages/d/duckdb/xmake.lua

```diff
@@ -0,0 +1,32 @@
+package("duckdb")
+    set_homepage("http://duckdb.org/")
+    set_description("DuckDB is an in-process SQL OLAP Database Management System")
+    set_license("MIT")
+
+    add_urls("https://github.com/duckdb/duckdb/releases/download/$(version)/libduckdb-src.zip",
+             "https://github.com/duckdb/duckdb.git")
+    add_versions("v0.10.0", "385e27aa67712813e4a07389465c4c5c45c431d97cddd35713b8a306d2a86f2d")
+
+    on_install("macosx", "linux", "windows", function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            set_languages("c++17")
```

Comment:
The README of this library says it needs c++11, why c++17?

Reply:
```
 49171 |     const S& format, basic_format_args<basic_printf_context_t<Char>> args) {
      |     ^~~~~
duckdb.cpp:49170:41: error: expected ')' before 'const'
49170 | inline std::basic_string<Char> vsprintf(
      |                                        ~^
      |                                         )
49171 |     const S& format, basic_format_args<basic_printf_context_t<Char>> args) {
      |     ~~~~~                                
duckdb.cpp:49188:1: warning: inline variables are only available with '-std=c++17' or '-std=gnu++17' [-Wc++17-extensions]
49188 | inline std::basic_string<Char> sprintf(const S& format, const Args&... args) {
      | ^~~~~~
duckdb.cpp:49186:23: error: parameter pack 'Args' must be at the end of the template parameter list
49186 | template <typename S, typename... Args,
      |                       ^~~~~~~~
duckdb.cpp:49188:40: error: expected unqualified-id before 'const'
49188 | inline std::basic_string<Char> sprintf(const S& format, const Args&... args) {
```

This is the amalgamation library, not the original source. So it might need higher version of cpp